diff --git a/src/angular-cli-files/models/webpack-configs/common.js b/src/angular-cli-files/models/webpack-configs/common.js
semver exclusivity >= 0.900.0
--- a/src/angular-cli-files/models/webpack-configs/common.js
+++ b/src/angular-cli-files/models/webpack-configs/common.js
@@ -385,6 +385,9 @@ function getCommonConfig(wco) {
         devtool: false,
         profile: buildOptions.statsJson,
         resolve: {
+            plugins: [
+                require('pnp-webpack-plugin'),
+            ],
             extensions: ['.ts', '.tsx', '.mjs', '.js'],
             symlinks: !buildOptions.preserveSymlinks,
             modules: [wco.tsConfig.options.baseUrl || projectRoot, 'node_modules'],
@@ -393,6 +396,9 @@ function getCommonConfig(wco) {
         resolveLoader: {
             symlinks: !buildOptions.preserveSymlinks,
             modules: loaderNodeModules,
+            plugins: [
+                require('pnp-webpack-plugin').moduleLoader(module),
+            ],
         },
         context: projectRoot,
         entry: entryPoints,
diff --git a/src/angular-cli-files/models/webpack-configs/common.js b/src/angular-cli-files/models/webpack-configs/common.js
semver exclusivity > 0.803.0 < 0.900.0
--- a/src/angular-cli-files/models/webpack-configs/common.js
+++ b/src/angular-cli-files/models/webpack-configs/common.js
@@ -32,10 +32,7 @@ function getCommonConfig(wco) {
     const { root, projectRoot, buildOptions, tsConfig } = wco;
     const { styles: stylesOptimization, scripts: scriptsOptimization } = buildOptions.optimization;
     const { styles: stylesSourceMap, scripts: scriptsSourceMap, vendor: vendorSourceMap, } = buildOptions.sourceMap;
-    const nodeModules = find_up_1.findUp('node_modules', projectRoot);
-    if (!nodeModules) {
-        throw new Error('Cannot locate node_modules directory.');
-    }
+    const nodeModules = undefined;
     // tslint:disable-next-line:no-any
     const extraPlugins = [];
     const entryPoints = {};
@@ -327,6 +324,9 @@ function getCommonConfig(wco) {
         devtool: false,
         profile: buildOptions.statsJson,
         resolve: {
+            plugins: [
+                require('pnp-webpack-plugin'),
+            ],
             extensions: ['.ts', '.tsx', '.mjs', '.js'],
             symlinks: !buildOptions.preserveSymlinks,
             modules: [wco.tsConfig.options.baseUrl || projectRoot, 'node_modules'],
@@ -334,6 +334,9 @@ function getCommonConfig(wco) {
         },
         resolveLoader: {
             modules: loaderNodeModules,
+            plugins: [
+                require('pnp-webpack-plugin').moduleLoader(module),
+            ],
         },
         context: projectRoot,
         entry: entryPoints,
diff --git a/src/protractor/index.js b/src/protractor/index.js
semver exclusivity < 0.900.0
--- a/src/protractor/index.js
+++ b/src/protractor/index.js
@@ -27,21 +27,14 @@ function runProtractor(root, options) {
 async function updateWebdriver() {
     // The webdriver-manager update command can only be accessed via a deep import.
     const webdriverDeepImport = 'webdriver-manager/built/lib/cmds/update';
-    const importOptions = [
-        // When using npm, webdriver is within protractor/node_modules.
-        `protractor/node_modules/${webdriverDeepImport}`,
-        // When using yarn, webdriver is found as a root module.
-        webdriverDeepImport,
-    ];
     let path;
-    for (const importOption of importOptions) {
-        try {
-            path = require.resolve(importOption);
-        }
-        catch (error) {
-            if (error.code !== 'MODULE_NOT_FOUND') {
-                throw error;
-            }
+    try {
+        const protractorPath = require.resolve('protractor');
+        path = require.resolve(webdriverDeepImport, { paths: [protractorPath] });
+    }
+    catch (error) {
+        if (error.code !== 'MODULE_NOT_FOUND') {
+            throw error;
         }
     }
     if (!path) {
